libs/sdk-backend-bear/src/backend/workspace/insights/insightFilterMerging.ts

```diff
@@ -0,0 +1,125 @@
+// (C) 2020 GoodData Corporation
+import {
+    IDateFilter,
+    ObjRef,
+    objRefToString,
+    filterObjRef,
+    isRelativeDateFilter,
+    relativeDateFilterValues,
+    isIdentifierRef,
+    isUriRef,
+    isAbsoluteDateFilter,
+    absoluteDateFilterValues,
+    newAbsoluteDateFilter,
+    uriRef,
+    newRelativeDateFilter,
+    DateAttributeGranularity,
+    IFilter,
+    isDateFilter,
+} from "@gooddata/sdk-model";
+import groupBy from "lodash/fp/groupBy";
+import isNil from "lodash/fp/isNil";
+import { partition } from "lodash/fp";
+import zip from "lodash/fp/zip";
+import { InvariantError } from "ts-invariant";
+
+/**
+ * Merges the filters provided with the filters already in an insight.
+ *
+ * @param insightFilters - filters specified by the insight
+ * @param addedFilters - filters to merge in
+ * @param objRefsToUris - function to return uris from refs (this is used to make sure the comparison makes sense for items with different types of refs)
+ * @internal
+ */
+export async function mergeFilters(
```

Comment:
This being async makes the calling code extremely complicated... It is needed, however, as we sometimes need to "normalize" the identifiers so that we can tell if an idRef and uriRef point to the same item. This requirement adds so much complexity I suggest we should revise if it is really necessary...

Reply:
Also, there is no way to use this in https://github.com/gooddata/gooddata-ui-sdk/blob/master/libs/sdk-model/src/execution/executionDefinition/index.ts#L94 without making this async for no real reason. Moreover, this cannot use any backend-specific logic... I have no idea how to work around this.